### PR TITLE
[FEATURE REQUEST] Respect app_provider values from capabilities

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App: https://github.com/owncloud/android/issues/4075

Library PR (if needed): https://github.com/owncloud/android-library/pull/577

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt [hunk 1]
```diff
@@ -23,16 +23,20 @@ package com.owncloud.android.data.appregistry
...
 import com.owncloud.android.data.appregistry.datasources.RemoteAppRegistryDataSource
+import com.owncloud.android.data.capabilities.datasources.LocalCapabilitiesDataSource
 import com.owncloud.android.domain.appregistry.AppRegistryRepository
 import com.owncloud.android.domain.appregistry.model.AppRegistryMimeType
 import kotlinx.coroutines.flow.Flow
 
 class OCAppRegistryRepository(
     private val localAppRegistryDataSource: LocalAppRegistryDataSource,
     private val remoteAppRegistryDataSource: RemoteAppRegistryDataSource,
+    private val localCapabilitiesDataSource: LocalCapabilitiesDataSource
```
Comment (manuelplazaspalacio) [inline owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt:34]: Pon una coma aquí.

File: owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt [hunk 2]
```diff
@@ -23,16 +23,20 @@ package com.owncloud.android.data.appregistry
...
 ) : AppRegistryRepository {
     override fun refreshAppRegistryForAccount(accountName: String) {
-        remoteAppRegistryDataSource.getAppRegistryForAccount(accountName).also {
+        val capabilities =  localCapabilitiesDataSource.getCapabilityForAccount(accountName)//remoteAppRegistryDataSource.getCapabilities(accountName)
```
Comment (manuelplazaspalacio) [inline owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt:37]: Ese código comentado sobra.

File: owncloudData/src/main/java/com/owncloud/android/data/appregistry/datasources/RemoteAppRegistryDataSource.kt [hunk 3]
```diff
@@ -25,7 +25,8 @@ import com.owncloud.android.domain.appregistry.model.AppRegistry
 
 interface RemoteAppRegistryDataSource {
     fun getAppRegistryForAccount(
-        accountName: String
+        accountName: String,
+        appUrl: String?
```
Comment (manuelplazaspalacio) [inline owncloudData/src/main/java/com/owncloud/android/data/appregistry/datasources/RemoteAppRegistryDataSource.kt:29]: Una coma ahí.

**Review by manuelplazaspalacio — CHANGES_REQUESTED** [review]
Some change here @Aitorbp

File: owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt (hunk 2 above)
Comment (Aitorbp) [reply owncloudData/src/main/java/com/owncloud/android/data/appregistry/OCAppRegistryRepository.kt]: Done

File: owncloudData/src/main/java/com/owncloud/android/data/appregistry/datasources/RemoteAppRegistryDataSource.kt (hunk 3 above)
Comment (Aitorbp) [reply owncloudData/src/main/java/com/owncloud/android/data/appregistry/datasources/RemoteAppRegistryDataSource.kt]: Done

**Review by manuelplazaspalacio — APPROVED** [review]

**Comment by jesmrec** [comment]
Fixed. 

Just tested in the attached demo server with different capability value. Let's give a try and get feedback.

**Pull request merged**